.github/workflows/build.yml

```
@@ -0,0 +1,188 @@
+          destination="${name}_${version}_manifest.json"
+
+          cp "$source" "$destination"
+          echo "filename=$destination" >> "$GITHUB_OUTPUT"
+      - uses: actions/upload-artifact@4cec3d8aa04e39d1a68397de0c4cd6fb9dce8ec1 # v4.6.1
+        with:
+          name: terraform-registry-manifest.json
+          path: ${{ steps.terraform-registry-manifest.outputs.filename }}
+          if-no-files-found: error
+
+  # Builds the product for all platforms except macOS.
+  #
+  # With `reproducible: report`, this job also reports whether the build is reproducible,
+  # but does not enforce it.
+  #
+  # Reference: https://github.com/hashicorp/actions-go-build/blob/main/README.md
+  build:
+    needs:
+      - get-go-version
+      - set-product-version
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      # TODO: Customize `matrix` for your provider. Compare to existing .goreleaser.yml.
+      # Verify expected Artifacts list for a workflow run.
+      matrix:
+        goos: [freebsd, windows, linux, darwin]
+        goarch: ["386", "amd64", "arm", "arm64"]
+        exclude:
+          - goos: freebsd
+            goarch: arm64
+          - goos: windows
+            goarch: arm64
+          - goos: windows
+            goarch: arm
+          - goos: darwin
+            goarch: 386
+          - goos: darwin
+            goarch: arm
+
+    name: Go ${{ needs.get-go-version.outputs.go-version }} ${{ matrix.goos }} ${{ matrix.goarch }} build
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: hashicorp/actions-go-build@v1
+        env:
+          CGO_ENABLED: 0
+          BASE_VERSION: ${{ needs.set-product-version.outputs.product-base-version }}
+          PRERELEASE_VERSION: ${{ needs.set-product-version.outputs.product-prerelease-version}}
+          METADATA_VERSION: ${{ env.METADATA }}
+        with:
+          # TODO: Customize `bin_name` for your provider. Compare to existing .goreleaser.yml.
+          # Protocol v6 providers should omit the `_x5` suffix.
+          bin_name: "${{ env.PKG_NAME }}_v${{ needs.set-product-version.outputs.product-version }}_x5"
+          product_name: ${{ env.PKG_NAME }}
+          product_version: ${{ needs.set-product-version.outputs.product-version }}
+          go_version: ${{ needs.get-go-version.outputs.go-version }}
+          os: ${{ matrix.goos }}
+          arch: ${{ matrix.goarch }}
+          reproducible: report
+          # TODO: Customize `go build` flags for your provider. Compare to existing .goreleaser.yml.
+          # TODO: Customize `ldflags` for your provider. Ensure that the import path and name in the -X flag
+          # are correct for your provider.
+          instructions: |
+            go build \
+              -o "$BIN_PATH" \
+              -trimpath \
+              -buildvcs=false \
+              -ldflags "-s -w -X 'main.version=${{ needs.set-product-version.outputs.product-version }}'"
```

I see that the provider declares a `version.ProviderVersion` variable here:

https://github.com/hashicorp/terraform-provider-tfe/blob/608dfddea25595430bfefe73e743974a8e3b1fb8/version/version.go#L9

If you'd like, you can set that at build-time with the following change. Curiously, I do not see this in [this provider's GoReleaser build command](https://github.com/hashicorp/terraform-provider-tfe/blob/608dfddea25595430bfefe73e743974a8e3b1fb8/.goreleaser.yml#L39), so I wonder if `version.ProductVersion` is being set at all in current releases.

```suggestion
              -ldflags "-s -w -X 'version.ProductVersion=${{ needs.set-product-version.outputs.product-version }}'"
```

---

Thanks for noticing this! Addressed in 0f08ecea. I'm not sure that this is still being used anywhere but I'd rather this PR be consistent with the old code.